hack/gen-crd.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -eu
+
+echo "Building controller-gen tool..."
+go build -o bin/controller-gen github.com/openshift/machine-api-operator/vendor/sigs.k8s.io/controller-tools/cmd/controller-gen
+
+dir=$(mktemp -d -t XXXXXXXX)
+echo $dir
+mkdir -p $dir/src/github.com/openshift/machine-api-operator/pkg/apis
+mkdir -p $dir/src/github.com/openshift/machine-api-operator/vendor
+
+cp -r pkg/apis/healthchecking $dir/src/github.com/openshift/machine-api-operator/pkg/apis/.
+cp -r vendor/github.com/openshift/cluster-api/pkg/apis/machine $dir/src/github.com/openshift/machine-api-operator/pkg/apis
+# Some dependencies need to be coppied as well. Othwerwise, controller-gen will complain about non-existing kind Unsupported
+cp -r vendor/k8s.io $dir/src/github.com/openshift/machine-api-operator/vendor/.
+cp -r vendor/github.com $dir/src/github.com/openshift/machine-api-operator/vendor/.
+
+cwd=$(pwd)
+pushd $dir/src/github.com/openshift/machine-api-operator
+GOPATH=$dir ${cwd}/bin/controller-gen crd --domain openshift.io
+popd
+
+echo "Coping generated CRDs"
```

Comment:
given
> The process is still partially manual. I.e. after the CRDs are updated, one still needs to check if all the necessary fields are present. E.g. add back additionalPrinterColumns field.
I think we should leave the copying step manual. Also we woudn't want to copy machineclass